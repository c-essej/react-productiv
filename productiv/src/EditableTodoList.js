import React from "react";
import EditableTodo from "./EditableTodo";

/** Show list of editable todos.
 *
 * Props:
 * - todos: array of [ todo, ... ]
 * - update(): fn to call to update a todo
 * - remove(): fn to call to remove a todo
 *
 * TodoApp -> EditableTodoList -> [ EditableTodo, ... ]
 */

function EditableTodoList({todos, update, remove}) {
  console.log(`Rendering editable todo list \n
      todos: ${todos} \n
      update:${update} \n
      remove: ${remove}`)
  return (
      <div>
        {todos.map(todo => <EditableTodo todo={todo}
        update={update}
        removeMe={()=>remove(todo.id)}
        key={todo.id}/>)}
      </div>
  );
}

export default EditableTodoList;


// remove={remove.bind(this,todo.id)}