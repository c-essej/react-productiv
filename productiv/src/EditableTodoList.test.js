import EditableTodoList from './EditableTodoList';
import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TEST_TODOS from './_testCommon';

describe("tests for EditableTodoList", function () {

  test("Renders without error", function () {
    render(<EditableTodoList todos={TEST_TODOS} />);
  });

  test("Renders correct number of EditableTodos", function () {
    const { container } = render(<EditableTodoList todos={TEST_TODOS} />);
    expect(container.querySelectorAll(".EditableTodo").length).toEqual(3);

  });

  it("matches snapshot", function () {
    const { container } = render(<EditableTodoList todos={TEST_TODOS} />);

  expect(container).toMatchSnapshot();
  })

});