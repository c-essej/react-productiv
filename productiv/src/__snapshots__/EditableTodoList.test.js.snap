// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for EditableTodoList matches snapshot 1`] = `
<div>
  <div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              test1
            </b>
             
            <small>
              Priority 1
            </small>
          </div>
          <div>
            <small>
              test1 description
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              test2
            </b>
             
            <small>
              Priority 2
            </small>
          </div>
          <div>
            <small>
              test2 description
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              test3
            </b>
             
            <small>
              Priority 3
            </small>
          </div>
          <div>
            <small>
              test3 description
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
